src/passes/ifStatementTempVarPostpender.ts

```diff
@@ -0,0 +1,192 @@
+    node.vStatements
+      .slice(0)
+      .reverse()
+      .map((child) => this.dispatchVisit(child, ast));
+  }
+
+  visitVariableDeclarationStatement(node: VariableDeclarationStatement, ast: AST): void {
+    node.vDeclarations.forEach((declaration) => this.liveVars.delete(declaration.name));
+    if (node.vInitialValue) {
+      this.dispatchVisit(node.vInitialValue, ast);
+    }
+  }
+
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    // ignore the function identifier
+    for (const arg of node.vArguments) {
+      this.dispatchVisit(arg, ast);
+    }
+  }
+
+  visitIfStatement(node: IfStatement, ast: AST): void {
+    ensureBothBranchesAreBlocks(node, ast);
+    assert(node.vFalseBody instanceof Block);
+
+    const liveVarsBefore = new Set(this.liveVars);
+    this.dispatchVisit(node.vTrueBody, ast);
+    const liveVarsAfterTrueBody = new Set(this.liveVars);
+
+    this.liveVars = new Set(liveVarsBefore);
+    let liveVarsAfterFalseBody = new Set<string>();
+
+    this.dispatchVisit(node.vFalseBody, ast);
+    liveVarsAfterFalseBody = new Set(this.liveVars);
+
+    this.liveVars = new Set(liveVarsBefore);
+    this.dispatchVisit(node.vCondition, ast);
+
+    this.liveVars = new Set([
+      ...this.liveVars,
+      ...liveVarsAfterFalseBody,
+      ...liveVarsAfterTrueBody,
+    ]);
+
+    const trueHasPathWithoutReturn = hasPathWithoutReturn(node.vTrueBody);
+    if (trueHasPathWithoutReturn) {
+      for (const liveVar of liveVarsBefore) {
+        const child = createCairoTempVar(liveVar, ast);
+        (node.vTrueBody as Block).appendChild(child);
+        ast.registerChild(child, node.vTrueBody);
+      }
+    }
+    const falseHasPathWithoutReturn = hasPathWithoutReturn(node.vFalseBody);
+
+    if (falseHasPathWithoutReturn) {
+      for (const liveVar of liveVarsBefore) {
+        const child = createCairoTempVar(liveVar, ast);
+        node.vFalseBody.appendChild(child);
+        ast.registerChild(child, node.vFalseBody);
+      }
+    }
+
+    if (falseHasPathWithoutReturn || trueHasPathWithoutReturn) {
+      const parent = node.parent;
+      const blockWithAfter = createBlock(
+        [node, ...[...liveVarsBefore].map((v) => createCairoTempVar(v, ast))],
+        ast,
+      );
+      ast.replaceNode(node, blockWithAfter, parent);
+    }
+  }
+}
+
+function ensureBothBranchesAreBlocks(node: IfStatement, ast: AST): void {
+  if (!(node.vTrueBody instanceof Block) && !(node.vTrueBody instanceof UncheckedBlock)) {
+    node.vTrueBody = createBlock([node.vTrueBody], ast);
+    ast.registerChild(node.vTrueBody, node);
+  }
+
+  if (node.vFalseBody === undefined) {
+    node.vFalseBody = createBlock([], ast);
+    ast.registerChild(node.vFalseBody, node);
+  } else if (!(node.vFalseBody instanceof Block) && !(node.vFalseBody instanceof UncheckedBlock)) {
+    node.vFalseBody = createBlock([node.vFalseBody], ast);
+    ast.registerChild(node.vFalseBody, node);
+  }
```

Comment:
Could be refactored to:
```
node.vFalseBody = ensureWrappedWithBlock(ast, node.node.vFalseBody)    ast.registerChild(node.vFalseBody, node);
```

Reply:
If yes, then I think it would be overkill, code repetition is so small

Reply:
Yea, new function, I was not entirely sure which solution I prefer as well. That's just a suggestion. It might be a useful utility function as there might be similar cases with other statements. And its slightly more readable